types/validation.go

```diff
@@ -138,6 +140,8 @@ func VerifyCommitLightTrusting(
 // for this commit, but there may be some intersection.
 //
 // This method DOES check all the signatures.
+//
+// CONTRACT: must run ValidateBasic() on commit before verifying and check vals (if untrusted).
```

Comment:
The validator set should be always trusted here. Otherwise, the methods without `Trusting` in the name are used...